browse/controllers/files/__init__.py

```diff
@@ -102,6 +102,17 @@ def withdrawn(arxiv_id: Identifier, had_specific_version: bool=False) -> Respons
                                          arxiv_id=arxiv_id),
                          404, headers)
 
+def not_public(arxiv_id: Identifier, had_specific_version: bool=False) -> Response:
+    """ Returned for pages whose source is encrypted/ made not public by the author.
+    Sets expire to one year, max allowed by RFC 2616"""
+    if had_specific_version:
+        headers = {'Cache-Control': 'max-age=31536000'}
+    else:
+        headers = {'Cache-Control': maxage(False)}
+    return make_response(
+        render_template("dissemination/not_public.html",arxiv_id=arxiv_id), 403, headers
```

Comment:
I'm tempted to do a status other than 403 to avoid ambiguity with crowdsec and real unauth .

Reply:
this was the behavior the legacy system had before but im happy to change it if it causes problems down the line

Reply:
How about `405`, or `451` instead? https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#4xx_client_errors

Reply:
403 is fine.

Reply:
405 isnt correct. 451 seems...extreme... and possibly slightly inaccurate.
i could see 200 being applicable because we returned the template OK or 403 because thats not a page they are allowed to access, there should be no links to the page on the site

Reply:
although maybe not `451`, as there's an IIS code with the same id, and it looks like it's handled differently in some MS browsers. https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#Internet_Information_Services

Reply:
Another alternative: `204`. Since there's no real error on the client side, we're just not returning content for the request.